### DISCUSSION TITLE
Automatic translation completed, no strings were updated.

### DISCUSSION THREAD

**Discussion by ghubz** [WeblateOrg/weblate#5884] [Q&A]
Hi,

I've enabled Google Translation V3 and it works fine when using it to translate individual words. 

However, when running the automatic translation feature under Tools, it returns "Automatic translation completed, no strings were updated.".

**To Reproduce**
Steps to reproduce the behaviour:

1. Go to any language in project
2. Click on tools > automatic translation
3. Select any settings and "Automatic translation source" to "Machine translation"
4. Click "Apply". Progress bar goes to 100% in a second and then shows the message mentioned above.

`WARNINGS:
?: (weblate.W033.Gerrit) Failure in loading VCS module for Gerrit: git: 'review' is not a git command. See 'git --help'.
 (1)
        HINT: https://docs.weblate.org/en/weblate-4.6/vcs.html
?: (weblate.W033.Subversion) Failure in loading VCS module for Subversion: git: 'svn' is not a git command. See 'git --help'.

The most similar command is
        serve
 (1)
        HINT: https://docs.weblate.org/en/weblate-4.6/vcs.html
`
**Versions:**
 * Weblate: 4.6
 * Django: 3.2
 * siphashc: 2.1
 * translate-toolkit: 3.3.4
 * lxml: 4.6.3
 * Pillow: 8.2.0
 * bleach: 3.3.0
 * python-dateutil: 2.8.1
 * social-auth-core: 4.1.0
 * social-auth-app-django: 4.0.0
 * django-crispy-forms: 1.11.2
 * oauthlib: 3.1.0
 * django-compressor: 2.4.1
 * djangorestframework: 3.12.4
 * django-filter: 2.4.0
 * django-appconf: 1.0.4
 * user-agents: 2.2.0
 * filelock: 3.0.12
 * setuptools: 56.0.0
 * jellyfish: 0.8.2
 * openpyxl: 3.0.7
 * celery: 5.0.5
 * kombu: 5.0.2
 * translation-finder: 2.9
 * weblate-language-data: 2021.4
 * html2text: 2020.1.16
 * pycairo: 1.20.0
 * pygobject: 3.40.1
 * diff-match-patch: 20200713
 * requests: 2.25.1
 * django-redis: 4.12.1
 * hiredis: 2.0.0
 * sentry_sdk: 1.0.0
 * Cython: 0.29.23
 * misaka: 2.1.1
 * GitPython: 3.1.14
 * borgbackup: 1.1.16
 * pyparsing: 2.4.7
 * pyahocorasick: 1.4.2
 * Python: 3.7.3
 * Git: 2.20.1
 * psycopg2-binary: 2.8.6
 * phply: 1.2.5
 * chardet: 4.0.0
 * ruamel.yaml: 0.17.4
 * tesserocr: 2.5.1
 * akismet: 1.1
 * boto3: 1.17.53
 * zeep: 4.0.0
 * aeidon: 1.9
 * iniparse: 0.5
 * Mercurial: 5.7.1
 * Redis server: 5.0.3
 * PostgreSQL server: 11.11
 * Database backends: django.db.backends.postgresql
 * Cache backends: default:RedisCache, avatar:FileBasedCache
 * Email setup: django.core.mail.backends.smtp.EmailBackend: localhost
 * OS encoding: filesystem=utf-8, default=utf-8
 * Celery: redis://localhost:6379, redis://localhost:6379, regular
 * Platform: Linux 4.19.0-14-cloud-amd64 (x86_64)

Thanks in advance!

**Comment by nijel** [comment]
Do you see anything it the logs? It should contain info in case the service could not be used.

**Reply by ghubz** [reply]
SYSLOG only shows:
```
INFO test/interface/pt: starting automatic translation 28edd45c-523b-4bb7-97bc-3707d4ef0087: mt: google-translate-api-v3
INFO test/interface/pt: completed automatic translation
```

Not sure if there's any other log. (weblate-translate.log doesn't contain much useful info)

**Reply by nijel** [reply]
Just checked the code and more info is shown only at debug level. Consider increasing `DEFAULT_LOGLEVEL` in settings to `DEBUG` to see more info.

**Reply by ghubz** [reply]
Thanks for your reply, however I already have logging set to DEBUG and restarted the uwsgi service (is anything else needed?)

**Reply by ghubz** [reply]
OK after restarting the whole machine the automatic translation is working. Apparently some service needed a restart other than the uwsgi one..!

**Reply by nijel** [reply]
Yes, the celery service needs restart on configuration change as well.

**Reply by ghubz** [reply]
Ah didn't know! Thanks!

**Reply by nijel** [reply]
Weblate will now warn on this as well: 6977023b457aeed2107f8d30155ac99eb4332de4
